certbot/compat/path.py

```diff
@@ -0,0 +1,31 @@
+"""This compat module wraps os.path to forbid some functions."""
+# pylint: disable=function-redefined
+from __future__ import absolute_import
+
+# First round of wrapping: we import statically all public attributes exposed by the os.path
+# module. This allows in particular to have pylint, mypy, IDEs be aware that most of os.path
+# members are available in certbot.compat.path.
+from os.path import *  # type: ignore  # pylint: disable=wildcard-import,unused-wildcard-import,redefined-builtin,os-module-forbidden
```

Comment:
I have the same concern here about `type: ignore` which I missed in my review of `certbot.compat.os`. I'm curious if there's anything we can do here. Disabling type checking on all functions in `os.path` doesn't seem ideal.

Reply:
See my response in https://github.com/certbot/certbot/pull/7242#discussion_r304701443